Sources/MongoSwift/ChangeStream.swift

```diff
@@ -0,0 +1,165 @@
+import mongoc
+
+/// A token used for manually resuming a change stream. Pass this to the `resumeAfter` or `startAfter` fields of
+/// `ChangeStreamOptions` to resume or start a change stream where a previous one left off.
+/// - SeeAlso: https://docs.mongodb.com/manual/changeStreams/#resume-a-change-stream
+public struct ChangeStreamToken: Codable {
+    private let resumeToken: Document
+
+    public init(resumeToken: Document) {
+        self.resumeToken = resumeToken
+    }
+
+    public init(from decoder: Decoder) throws {
```

Comment:
why do we need a custom implementation of this?

Reply:
The `resumeToken` within `ChangeStreamToken` wasn't getting decoded (coding keys error), so @patrickfreed suggested using a decoder init.

Reply:
There was a decoding error while updating `resumeToken` from the `_id` field in `next`.
`"No value associated with key CodingKeys(stringValue: \"resumeToken\", intValue: nil) (\"resumeToken\").", underlyingError: nil))`
So @patrickfreed  suggested using an decoder init so the `resumeToken` gets initialized as a document.

Reply:
Right. The `ChangeStreamToken` type doesn't actually care about the fields present in it (and they're subject to change according to server version), so the `Codable` conformance is just deferred to the underlying document being wrapped. There isn't actually a `resumeToken` field (as per the error received when relying on the synthesized conformance) in the `_id` field of a change stream event.